executor/join.go

```diff
@@ -443,84 +447,84 @@ func (e *HashJoinExec) waitJoinWorkersAndCloseResultChan() {
 		// Concurrently handling unmatched rows from the hash table at the tail
 		for i := uint(0); i < e.concurrency; i++ {
 			var workerID = i
-			e.worker.RunWithRecover(func() { e.handleUnmatchedRowsFromHashTable(workerID) }, e.handleJoinWorkerPanic)
+			e.worker.RunWithRecover(func() { e.probeWorkers[workerID].handleUnmatchedRowsFromHashTable() }, e.handleJoinWorkerPanic)
 		}
 		e.worker.Wait()
 	}
 	close(e.joinResultCh)
 }
 
-func (e *HashJoinExec) runJoinWorker(workerID uint, probeKeyColIdx, probeNAKeyColIdx []int) {
+func (w *probeWorker) runJoinWorker(probeKeyColIdx, probeNAKeyColIdx []int) {
 	probeTime := int64(0)
-	if e.stats != nil {
+	if w.stats != nil {
 		start := time.Now()
 		defer func() {
 			t := time.Since(start)
-			atomic.AddInt64(&e.stats.probe, probeTime)
-			atomic.AddInt64(&e.stats.fetchAndProbe, int64(t))
-			e.stats.setMaxFetchAndProbeTime(int64(t))
+			atomic.AddInt64(&w.stats.probe, probeTime)
+			atomic.AddInt64(&w.stats.fetchAndProbe, int64(t))
+			w.stats.setMaxFetchAndProbeTime(int64(t))
 		}()
 	}
 
 	var (
 		probeSideResult *chunk.Chunk
 		selected        = make([]bool, 0, chunk.InitialCapacity)
 	)
-	ok, joinResult := e.getNewJoinResult(workerID)
+	ok, joinResult := w.getNewJoinResult()
 	if !ok {
 		return
 	}
 
 	// Read and filter probeSideResult, and join the probeSideResult with the build side rows.
 	emptyProbeSideResult := &probeChkResource{
-		dest: e.probeSideTupleFetcher.probeResultChs[workerID],
+		dest: w.probeResultCh,
 	}
 	hCtx := &hashContext{
-		allTypes:    e.probeTypes,
+		allTypes:    w.probeTypes,
 		keyColIdx:   probeKeyColIdx,
 		naKeyColIdx: probeNAKeyColIdx,
 	}
 	for ok := true; ok; {
-		if e.finished.Load() {
+		if w.finished.Load() {
 			break
 		}
 		select {
-		case <-e.closeCh:
+		case <-w.closeCh:
 			return
-		case probeSideResult, ok = <-e.probeSideTupleFetcher.probeResultChs[workerID]:
+		case probeSideResult, ok = <-w.probeResultCh:
 		}
 		failpoint.Inject("ConsumeRandomPanic", nil)
 		if !ok {
 			break
 		}
 		start := time.Now()
-		if e.useOuterToBuild {
-			ok, joinResult = e.join2ChunkForOuterHashJoin(workerID, probeSideResult, hCtx, e.probeWorkers[workerID].rowContainerForProbe, joinResult)
+		if w.useOuterToBuild {
+			ok, joinResult = w.join2ChunkForOuterHashJoin(probeSideResult, hCtx, joinResult)
 		} else {
-			ok, joinResult = e.join2Chunk(workerID, probeSideResult, hCtx, e.probeWorkers[workerID].rowContainerForProbe, joinResult, selected)
+			ok, joinResult = w.join2Chunk(probeSideResult, hCtx, w.rowContainerForProbe, joinResult, selected)
```

Comment:
Why `join2ChunkForOuterHashJoin` remove the arg `e.probeWorkers[workerID].rowContainerForProbe` but `join2Chunk` not?